Assets/LeapMotionModules/UI/Scripts/ColliderUtil.cs

```diff
@@ -0,0 +1,281 @@
+  //GENRAL HELPER
+
+  /* Returns whether or not a given vector is within a given radius to a center point
+   */
+  public static bool withinDistance(Vector3 point, Vector3 center, float radius) {
+    return (point - center).sqrMagnitude < radius * radius;
+  }
+
+  public static bool withinDistance(Vector3 pointToOrigin, float radius) {
+    return pointToOrigin.sqrMagnitude < radius * radius;
+  }
+
+  /* Given a point and a center, return a new point that is along the axis created by the 
+   * two vectors, and is a given distance from the center
+   */
+  public static Vector3 getPointAtDistance(Vector3 point, Vector3 center, float distance) {
+    return (point - center).normalized * distance + center;
+  }
+
+
+  /* Given a collider and a position relative to the colliders transform, return whether or not the 
+   * position lies within the collider.  Can also provide the allowed distance tolerance
+   */
+  public static bool isInsideCollider(this Collider collider, Vector3 localPosition, float tolerance = 0.0f) {
+    SphereCollider sphereCollider = collider as SphereCollider;
+    if (sphereCollider != null) {
+      return sphereCollider.isPointInside(localPosition, tolerance);
+    }
+    BoxCollider boxCollider = collider as BoxCollider;
+    if (boxCollider != null) {
+      return boxCollider.isPointInside(localPosition, tolerance);
+    }
+    CapsuleCollider capsuleCollider = collider as CapsuleCollider;
+    if (capsuleCollider != null) {
+      return capsuleCollider.isPointInside(localPosition, tolerance);
+    }
+    Debug.LogWarning("ColliderUtil does not work for Mesh Colliders!");
+    return false;
+  }
+
+  /* Given a collider and a position reltative to the colliders transform, return the point on the
+   * surface of the collider closest to the position
+   */
+  public static Vector3 closestPointOnSurface(this Collider collider, Vector3 localPosition, float extrude = 0.0f) {
+    SphereCollider sphereCollider = collider as SphereCollider;
+    if (sphereCollider != null) {
+      return sphereCollider.closestPointOnSurface(localPosition, extrude);
+    }
+    BoxCollider boxCollider = collider as BoxCollider;
+    if (boxCollider != null) {
+      return boxCollider.closestPointOnSurface(localPosition, extrude);
+    }
+    CapsuleCollider capsuleCollider = collider as CapsuleCollider;
+    if (capsuleCollider != null) {
+      return capsuleCollider.closestPointOnSurface(localPosition, extrude);
+    }
+    Debug.LogWarning("ColliderUtil does not work for Mesh Colliders!");
+    return Vector3.zero;
+  }
+
+  /* Given a GameObject and a position in global space, return whether or not the position lies within
+   * one of the colliders on the GameObject or one of it's children.  Can also provide the allowed 
+   * distance tolerance
+   */
+  public static bool isInsideColliders(GameObject obj, Vector3 globalPosition, float tolerance = 0.0f) {
+    obj.GetComponentsInChildren<Collider>(false, _colliderList);
```

Comment:
I would recommend allowing this method to take in a list of colliders instead of a gameObject, that way we can avoid doing a GetComponent every frame